### PR TITLE
Order settings mutation and query

### PR DESCRIPTION
**Pull request by tomaszszymanski129**
I want to merge this change because it adds a query and mutation for orderSettings.

Related ticket: [SALEOR-1418](https://app.clickup.com/t/2549495/SALEOR-1418)

_This is a PR to feature branch._

# Impact

* [ ] New migrations
* [ ] New/Updated API fields or mutations
* [ ] Deprecated API fields or mutations
* [ ] Removed API types, fields, or mutations
* [ ] Documentation needs to be updated

# Pull Request Checklist

* [ ] Privileged queries and mutations are guarded by proper permission checks
* [ ] Database queries are optimized and the number of queries is constant
* [ ] Database migration files are up to date
* [ ] The changes are tested
* [ ] GraphQL schema and type definitions are up to date
* [ ] Changes are mentioned in the changelog

### CODE REVIEW

**Comment by patrys** [comment]
Task linked: [SALEOR-1418 Add query/mutation to save order settings](https://app.clickup.com/t/2549495/SALEOR-1418)

File: saleor/graphql/order/types.py [hunk 1]
```diff
@@ -37,6 +38,13 @@ class OrderEventOrderLineObject(graphene.ObjectType):
     item_name = graphene.String(description="The variant name.")
 
 
+class SiteSettings(CountableDjangoObjectType):
+    class Meta:
+        only_fields = ["automatically_confirm_all_new_orders"]
+        description = "Checkout object."
```
Comment (IKarbowiak) [inline saleor/graphql/order/types.py:44]: The description must be changed. 👮‍♀️

File: saleor/graphql/order/mutations/orders.py [hunk 2]
```diff
@@ -483,3 +484,32 @@ def perform_mutation(cls, _root, info, amount, **data):
...
+
+class OrderUpdateSettingsInput(graphene.InputObjectType):
+    automatically_confirm_all_new_orders = graphene.Boolean(
+        required=True,
+        description="When enabled, all new orders from checkout "
+        "will be marked as unconfirmed",
+    )
+
+
+class OrderUpdateSettings(ModelMutation):
```
Comment (IKarbowiak) [inline saleor/graphql/order/mutations/orders.py:497]: To be consistent with others mutation should be named: `OrderSettingsUpdate`.

File: saleor/graphql/order/tests/test_order.py [hunk 3]
```diff
@@ -450,6 +450,59 @@ def test_draft_order_query(staff_api_client, permission_manage_orders, orders):
     assert len(edges) == Order.objects.drafts().count()
 
 
+def test_order_update_settings(
```
Comment (IKarbowiak) [inline saleor/graphql/order/tests/test_order.py:453]: We should also add tests for running mutation by the app, and by the user without permission.

File: saleor/graphql/core/types/common.py [hunk 4]
```diff
@@ -121,6 +122,10 @@ class MenuError(Error):
     code = MenuErrorCode(description="The error code.", required=True)
 
 
+class OrderSettingsError(Error):
```
Comment (fowczarek) [inline saleor/graphql/core/types/common.py:125]: You should add `OrderSettingsError` to `SALEOR_ERROR_CODE_ENUMS` in `saleor/saleor/graphql/core/utils/error_codes.py`

File: saleor/graphql/order/mutations/orders.py [hunk 5]
```diff
@@ -483,3 +484,32 @@ def perform_mutation(cls, _root, info, amount, **data):
         if transaction.kind == TransactionKind.REFUND:
             order_refunded(order, info.context.user, amount, payment)
         return OrderRefund(order=order)
+
+
+class OrderUpdateSettingsInput(graphene.InputObjectType):
+    automatically_confirm_all_new_orders = graphene.Boolean(
+        required=True,
+        description="When enabled, all new orders from checkout "
+        "will be marked as unconfirmed",
```
Comment (fowczarek) [inline saleor/graphql/order/mutations/orders.py:493]: :police_officer: 
```suggestion
        "will be marked as unconfirmed.",
```

File: saleor/graphql/order/schema.py [hunk 6]
```diff
@@ -64,6 +65,7 @@ class OrderQueries(graphene.ObjectType):
             "homepage (at the moment it only contains order-events)."
         ),
     )
+    order_settings = graphene.Field(SiteSettings)
```
Comment (fowczarek) [inline saleor/graphql/order/schema.py:68]: Missing description.

File: saleor/graphql/order/types.py [hunk 7]
```diff
@@ -37,6 +38,13 @@ class OrderEventOrderLineObject(graphene.ObjectType):
     item_name = graphene.String(description="The variant name.")
 
 
+class SiteSettings(CountableDjangoObjectType):
```
Comment (fowczarek) [inline saleor/graphql/order/types.py:41]: Maybe this type should be named `OrderSettings`?

File: saleor/graphql/order/mutations/orders.py (hunk 2 above)
Comment (fowczarek) [inline saleor/graphql/order/mutations/orders.py:497]: I think that we should use BaseMutation here.

**Review by IKarbowiak — APPROVED** [review]

**Review by fowczarek — APPROVED** [review]

File: saleor/graphql/order/mutations/orders.py [hunk 8]
```diff
@@ -483,3 +483,35 @@ def perform_mutation(cls, _root, info, amount, **data):
         if transaction.kind == TransactionKind.REFUND:
             order_refunded(order, info.context.user, amount, payment)
         return OrderRefund(order=order)
+
+
+class OrderSettingsUpdateInput(graphene.InputObjectType):
+    automatically_confirm_all_new_orders = graphene.Boolean(
+        required=True,
+        description="When enabled, all new orders from checkout "
+        "will be marked as unconfirmed.",
```
Comment (maarcingebala) [inline saleor/graphql/order/mutations/orders.py:492]: "will be marked as unconfirmed" -> is this correct? This setting is about making them confirmed. I would also say what will happen if the setting is disabled.

File: saleor/graphql/order/types.py [hunk 9]
```diff
@@ -37,6 +38,13 @@ class OrderEventOrderLineObject(graphene.ObjectType):
     item_name = graphene.String(description="The variant name.")
 
 
+class OrderSettings(CountableDjangoObjectType):
```
Comment (maarcingebala) [inline saleor/graphql/order/types.py:41]: I think it should be part of the settings module since it uses site settings model underneath.

Comment (maarcingebala) [reply saleor/graphql/order/types.py:41]: I meant the `saleor/graphql/shop/types.py` module. If we moved this type there, also the `OrderSettingsUpdate` could be moved to `shop/mutations`. Then we'll have all "settings" mutations in one place.

**Review by maarcingebala — APPROVED** [review]

**Pull request merged**